libtenzir/builtins/contexts/lookup_table.cpp

```diff
@@ -481,19 +585,54 @@ class ctx final : public virtual context {
     //   [{key: key, value: value}, ...]
     auto builder = flatbuffers::FlatBufferBuilder{};
     auto value_offsets = std::vector<flatbuffers::Offset<fbs::Data>>{};
-    value_offsets.reserve(context_entries.size());
+    value_offsets.reserve(context_entries.size()); // TODO: store as variable
+    auto now = time::clock::now();
     for (const auto& [key, value] : context_entries) {
+      if ((value.update_timeout && value.update_timeout < now)
+          || (value.create_timeout && value.create_timeout < now)) {
+        continue;
+      }
       auto field_offsets
         = std::vector<flatbuffers::Offset<fbs::data::RecordField>>{};
-      field_offsets.reserve(2);
+      u_int8_t to_reserve = 2;
+      if (value.create_timeout) {
+        to_reserve += 1;
+      }
+      if (value.update_timeout) {
+        to_reserve += 1;
+      }
+      field_offsets.reserve(to_reserve);
       const auto key_key_offset = builder.CreateSharedString("key");
       const auto key_value_offset = pack(builder, key.to_original_data());
       field_offsets.emplace_back(fbs::data::CreateRecordField(
         builder, key_key_offset, key_value_offset));
-      const auto value_key_offset = builder.CreateSharedString("value");
-      const auto value_value_offset = pack(builder, value);
+      auto value_key_offset = builder.CreateSharedString("data");
```

Comment:
```suggestion
      auto value_key_offset = builder.CreateSharedString("value");
```

For backwards compat.